Improved render

Avenger-Bagnarol opened the pull request:
Modified render.py to allow distinction of cells with different stiffness. 
Works with smooth option.

pmadhikar commented on scripts/render.py:
```diff
@@ -5,7 +5,9 @@
-sys.path.append("/home/pranav/dev/celldiv/scripts")
+#Default: violet stiff (healthy), orange soft (ill)
+
+sys.path.append("/Users/Torsa_Legend/Desktop/ROBE TESI/Progr/scripts")
```
Ah, we have to change this line. It shouldn't be required anymore since the compilation should add it to `PYTHONPATH`

pmadhikar commented on scripts/render.py:
```diff
@@ -120,20 +126,36 @@
+lamp.type = 'SUN'  # in ['POINT', 'SUN', 'SPOT', 'HEMI', 'AREA']
+lamp.distance = 100
+
+#mio
```
Please remove Italian comments.

pmadhikar commented on scripts/render.py:
```diff
@@ -183,7 +249,11 @@
+            bpy.ops.object.delete()                  # delete mesh...
+            
+            #roba mia
+            bpy.ops.object.select_pattern(pattern='scellObject')
```
What if there are no soft cells?

pmadhikar commented on scripts/render.py:
```diff
@@ -142,21 +164,49 @@
+              f = np.vstack(f)   
+              faces = []
+              for mi in range(int(len(f)/192)):
                 for row in firstfaces:
```
The geometry definition should probably be moved into a function call...

Avenger-Bagnarol commented:
1) I didn't delete anything, just modified render. I can't tell which file
you see missing, and why.
2) I deleted three comments. Should I make another pull request just for
that?
3) If there are no soft cell, as far as I know, the selection is set to
Null and it doesn't delete anything. Should I add boolean guards?
4) What exactly do you want in a new function?
5) I am not sure that "sys.path.append("/Users/Torsa_Legend/Desktop/ROBE
TESI/Progr/scripts")" is not required. I invite you to generalize it using
PYTHONPATH.

Il giorno mar 14 mag 2019 alle ore 11:53 pmadhikar <notifications@github.com>
ha scritto:

> *@pmadhikar* requested changes on this pull request.
>
> Why did you delete a file?
> ------------------------------
>
> In scripts/render.py
> <https://github.com/SoftSimu/CellSim3D/pull/5#discussion_r283716841>:
>
> > @@ -183,7 +249,11 @@
>              bpy.ops.render.render(write_still=True)  # render to file
>
>              bpy.ops.object.select_pattern(pattern='cellObject')
> -            bpy.ops.object.delete()                                     # delete mesh...
> +            bpy.ops.object.delete()                  # delete mesh...
> +
> +            #roba mia
> +            bpy.ops.object.select_pattern(pattern='scellObject')
>
> What if there are no soft cells?
> ------------------------------
>
> In scripts/render.py
> <https://github.com/SoftSimu/CellSim3D/pull/5#discussion_r283717705>:
>
> > -            for mi in range(int(len(f)/192)):
> +            f0=[]
> +            f1=[]
> +            if sum(th.currTypes) > 0.1: #there is at least one cell with nonzero index
> +                for cc in range(len(th.currTypes)):
> +                   if (th.currTypes[cc]):
> +                      f1.append(f[cc])
> +                   else:
> +                      f0.append(f[cc])
> +
> +
> +            if len(f0) > 0:
> +              f=f0
> +              f = np.vstack(f)
> +              faces = []
> +              for mi in range(int(len(f)/192)):
>                  for row in firstfaces:
>
> The geometry definition should probably be moved into a function call...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SoftSimu/CellSim3D/pull/5?email_source=notifications&email_token=ALVGTTJTGA6GFI7AHT7QNS3PVKD2VA5CNFSM4HMYDGPKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBYRHJ3A#pullrequestreview-237139180>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALVGTTNKDWISPBYZXMQRAD3PVKD2VANCNFSM4HMYDGPA>
> .
>

pmadhikar commented:
You don't need to make a new pull request. This one will update automatically whenever you push to the same branch. 
The file LIECENSE/jsoncpp is missing.
